packages/dev/core/src/AudioV2/abstractAudio/spatialAttachers/spatialAudioMeshAttacher.ts

```diff
@@ -0,0 +1,33 @@
+import { Vector3 } from "../../../Maths/math.vector";
+import type { AbstractMesh } from "../../../Meshes/abstractMesh";
+import type { Nullable } from "../../../types";
+import type { _SpatialAudioAttacherComponent } from "../components/spatialAudioAttacherComponent";
+import type { _AbstractSpatialAudioAttacher } from "./abstractSpatialAudioAttacher";
+import { _SpatialAudioTransformNodeAttacher } from "./spatialAudioTransformNodeAttacher";
+
+/**
+ * NB: This function is async so it can use a dynamic import in the future if needed.
+ * @internal
+ * */
+export async function _CreateSpatialAudioMeshAttacherAsync(attacherComponent: _SpatialAudioAttacherComponent): Promise<_AbstractSpatialAudioAttacher> {
+    return new _SpatialAudioMeshAttacher(attacherComponent);
+}
+
+/** @internal */
+class _SpatialAudioMeshAttacher extends _SpatialAudioTransformNodeAttacher {
+    protected override _transformNode: Nullable<AbstractMesh>;
+
+    /** @internal */
+    public constructor(attacherComponent: _SpatialAudioAttacherComponent) {
+        super(attacherComponent);
+    }
+
+    protected override get _attachedPosition(): Vector3 {
+        return this._transformNode?.getBoundingInfo().boundingSphere.centerWorld ?? Vector3.ZeroReadOnly;
```

Comment:
if this is the only difference between a mesh and transform node, why not have the function look like this for both?
```suggestion
        return (this._transformNode as AbstractMesh)?.getBoundingInfo?.().boundingSphere.centerWorld || this._transformNode?.absolutePosition || Vector3.ZeroReadOnly;
```

Reply:
In the case of meshes, I want the user to be able to choose which algorithm to use for the spatial audio positioning. Currently, if the user sets the `attachedTransformNode` property then the node's position will be used for the spatial audio positioning, and if the user sets the `attachedMesh` property then the node's bounding box will be used for the spatial audio positioning.

If we detect the type automatically, then the user is forced to use the bounding box method for all meshes, even if they want the transform node position to be used instead.

We could add a flag to switch between the two algorithms instead of using 2 classes, but the `_SpatialAudioMeshAttacher` class is not huge, so it wouldn't reduce the size much I don't think.

Reply:
got it, let's keep it this way